controllers/backstage_route.go

```diff
@@ -56,9 +56,57 @@ func (r *BackstageReconciler) applyBackstageRoute(ctx context.Context, backstage
 		}
 	}
 
+	r.applyRouteParamsFromCR(route, backstage)
+
 	err = r.Create(ctx, route)
 	if err != nil {
 		return fmt.Errorf("failed to create backstage route, reason: %s", err)
 	}
 	return nil
 }
+
+func (r *BackstageReconciler) applyRouteParamsFromCR(route *openshift.Route, backstage bs.Backstage) {
+	if backstage.Spec.Application == nil || backstage.Spec.Application.Route == nil {
+		return // Nothing to override
+	}
+	routeCfg := backstage.Spec.Application.Route
+	if len(routeCfg.Host) > 0 {
+		route.Spec.Host = routeCfg.Host
+	}
+	if len(routeCfg.Subdomain) > 0 {
+		route.Spec.Subdomain = routeCfg.Subdomain
+	}
+	if routeCfg.TLS == nil {
+		return
+	}
+	if route.Spec.TLS == nil {
+		route.Spec.TLS = &openshift.TLSConfig{
```

Comment:
Not sure why all fields (including default ones) are being set here, especially if they are set below again if `route.Spec.TLS != nil`. Would just something like below be sufficient?

```go
if route.Spec.TLS == nil {
  route.Spec.TLS = &openshift.TLSConfig{}
}
// then set the rest of the fields from `routeCfg`
if routeCfg.TLS.Certificate != "" {
  route.Spec.TLS.Certificate = routeCfg.TLS.Certificate
}
// -- Truncated --
```

Reply:
Actually it is more efficient to use the constructor to initialize the data, and it also avoids the if statement checks later on.